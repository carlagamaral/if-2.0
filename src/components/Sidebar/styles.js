export default {
	bmBurgerBars: {
		background: '#373a47',
		visibility: 'hidden',
	},
	bmBurgerBarsHover: {
		background: '#a90000',
	},
	bmBurgerButton: {
		position: 'relative',
		width: '36px',
	},
	bmItem: {
		display: 'inline-block',
		outline: 'none',
	},
	bmItemList: {
		color: 'white',
		padding: '0.8em',
	},
	bmMenu: {
		background: '#3eb0b8',
		fontSize: '1.15em',
		overflowY: 'hidden',
		padding: '1em 1.5em 0',
	},
	bmMenuWrap: {
		height: '100%',
		position: 'fixed',
		top: '0',
	},
	bmMorphShape: {
		fill: '#373a47',
	},
	bmOverlay: {
		background: 'rgba(0, 0, 0, 0.3)',
		top: 0,
	},
};
