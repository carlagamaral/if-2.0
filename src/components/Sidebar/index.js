import React from 'react';
import PropTypes from 'prop-types';
import { slide as Menu } from 'react-burger-menu';
import styles from './styles';

const Sidebar = ({ children, isOpen, onOpen, onClose, width }) => {
	return (
		<Menu
			isOpen={isOpen}
			onClose={onClose}
			onOpen={onOpen}
			styles={styles}
			width={width}
		>
			{children}
		</Menu>
	);
};

Sidebar.propTypes = {
	burgerIcon: PropTypes.any,
	children: PropTypes.arrayOf(PropTypes.element),
	isOpen: PropTypes.bool.isRequired,
	onClose: PropTypes.func,
	onOpen: PropTypes.func,
	width: PropTypes.string.isRequired,
};

Sidebar.defaultProps = {
	burgerIcon: undefined, // eslint-disable-line no-undefined
	children: null,
	onClose: () => {},
	onOpen: () => {},
};

export default Sidebar;
