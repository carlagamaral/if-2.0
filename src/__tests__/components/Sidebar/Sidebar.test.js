/* eslint-disable function-paren-newline */
import 'jest-dom/extend-expect'
import { cleanup, waitForElement, render, fireEvent } from 'react-testing-library'
import React from 'react'
import Sidebar from '../../../components/Sidebar'

describe( 'Sidebar tests ', () => {
  const divId = 'divSidebar'

  afterEach( () => {
    cleanup()
    jest.clearAllMocks()
  } )

  test( 'Sidebar renders it children', async() => {
    const { container, getByTestId } = render(
      <Sidebar isOpen={ false } >
        <div data-testid={ divId }>
          Conteúdo Sidebar
        </div>
      </Sidebar>
    )
    const div = await waitForElement( () => getByTestId( divId ) )

    expect( container ).toMatchSnapshot()
    expect( div ).toBeInTheDocument()
  } )

  test( 'Sidebar renders the burgerIcon as a custom component', async() => {
    const dataTestIdBurgerIcon = 'burgerIcon'
    const { getByTestId } = render(
      <Sidebar
        burgerIcon={ <div data-testid={ dataTestIdBurgerIcon } /> }
        isOpen={ false }
      >
        <div data-testid={ divId }>
          Conteúdo Sidebar
        </div>
      </Sidebar>
    )

    const burgerIcon = await waitForElement( () => getByTestId( dataTestIdBurgerIcon ) )

    expect( burgerIcon ).toBeInTheDocument()
  } )

  test( 'Sidebar calls the open/close handler when interacting with it', () => {
    const openHandler = jest.fn()
    const closeHandler = jest.fn()

    const { getByText } = render(
      <Sidebar
        isOpen={ false }
        onClose={ closeHandler }
        onOpen={ openHandler }
      >
        <div data-testid={ divId }>
          Conteúdo Sidebar
        </div>
      </Sidebar>
    )

    const openMenuButton = getByText( 'Open Menu' )
    const closeMenuButton = getByText( 'Close Menu' )

    fireEvent.click( openMenuButton )
    expect( openHandler ).toHaveBeenCalled()

    fireEvent.click( closeMenuButton )
    expect( closeHandler ).toHaveBeenCalled()

    fireEvent.click( openMenuButton )
    expect( openHandler ).toHaveBeenCalledTimes( 2 )
  } )

  test( 'Sidebar is rendered with the right width', () => {
    const width = '600px'

    const { container } = render(
      <Sidebar
        isOpen={ false }
        width={ width }
      >
        <div data-testid={ divId }>
          Conteúdo Sidebar
        </div>
      </Sidebar>
    )

    // eslint-disable-next-line no-underscore-dangle
    expect( container.firstChild.childNodes[ 1 ].style._values.width ).toEqual( width )
  } )
} )
