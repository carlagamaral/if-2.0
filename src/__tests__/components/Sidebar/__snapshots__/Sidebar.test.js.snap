// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar tests  Sidebar renders it children 1`] = `
<div>
  <div>
    <div
      class="bm-overlay"
      style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
    />
    <div
      class="bm-menu-wrap"
      id=""
      style="position: fixed; z-index: 1100; width: 1000px; height: 100%; transform: translate3d(-100%, 0, 0); transition: all 0.5s;"
    >
      <div
        class="bm-menu"
        style="height: 100%; box-sizing: border-box; overflow: auto; background: rgb(55, 58, 71); font-size: 1.15em; overflow-y: hidden; padding: 1em 1.5em 0px;"
      >
        <nav
          class="bm-item-list"
          style="height: 100%; color: white; padding: 0.8em;"
        >
          <div
            class="bm-item"
            data-testid="divSidebar"
            style="display: inline-block;"
            tabindex="-1"
          >
            Conte√∫do Sidebar
          </div>
        </nav>
      </div>
      <div>
        <div
          class="bm-cross-button"
          style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
        >
          <span
            style="position: absolute; top: 6px; right: 14px;"
          >
            <span
              class="bm-cross"
              style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg); background: rgb(189, 195, 199);"
            />
            <span
              class="bm-cross"
              style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg); background: rgb(189, 195, 199);"
            />
          </span>
          <button
            style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
            tabindex="-1"
          >
            Close Menu
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        class="bm-burger-button"
        style="z-index: 1000; height: 30px; left: 15px; position: relative; top: 25px; width: 36px;"
      >
        <span>
          <span
            class="bm-burger-bars"
            style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1; background: rgb(55, 58, 71);"
          />
          <span
            class="bm-burger-bars"
            style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1; background: rgb(55, 58, 71);"
          />
          <span
            class="bm-burger-bars"
            style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1; background: rgb(55, 58, 71);"
          />
        </span>
        <button
          style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
        >
          Open Menu
        </button>
      </div>
    </div>
  </div>
</div>
`;
